### ISSUE TITLE
Annoyance: Transaction extension tuple can't be more than 12 elements, and tuple of tuple changes order of inherited implication and is not discoverable with metadata

### ISSUE THREAD

**Issue by gui1117** [paritytech/polkadot-sdk#6569]
TLDR: Transaction extension tuple can't be more than 12 elements, and tuple of tuple changes order of inherited implication, and the order is not discoverable with metadata because it only contains a vector of transaction extension.

### Description:

The transaction extensions are define with a tuple of transaction extension, and is part of the extrinsic that is part of a block.
```rust
pub type TxExtension = (
	frame_system::CheckNonZeroSender<Runtime>,
	frame_system::CheckSpecVersion<Runtime>,
	frame_system::CheckTxVersion<Runtime>,
	frame_system::CheckGenesis<Runtime>,
	frame_system::CheckEra<Runtime>,
	frame_system::CheckNonce<Runtime>,
	frame_system::CheckWeight<Runtime>,
	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
	cumulus_primitives_storage_weight_reclaim::StorageWeightReclaim<Runtime>,
	frame_metadata_hash_extension::CheckMetadataHash<Runtime>,
);

/// Unchecked extrinsic type as expected by this runtime.
pub type UncheckedExtrinsic =
	generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, TxExtension>;
```

The block requires among other traits: `Eq`, `PartialEq`, `Debug`.

Those specific traits are implemented on tuple of at most 12 elements: https://doc.rust-lang.org/core/primitive.tuple.html#trait-implementations-1

To fit into 12 elements we may want to group some into tuples like:
```rust
pub type TxExtension = (
	(
		frame_system::CheckNonZeroSender<Runtime>,
		frame_system::CheckSpecVersion<Runtime>,
		frame_system::CheckTxVersion<Runtime>,
		frame_system::CheckGenesis<Runtime>,
		frame_system::CheckEra<Runtime>,
	),
	frame_system::CheckNonce<Runtime>,
	frame_system::CheckWeight<Runtime>,
	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
	cumulus_primitives_storage_weight_reclaim::StorageWeightReclaim<Runtime>,
	frame_metadata_hash_extension::CheckMetadataHash<Runtime>,
);
```

But this changes the inherited implication passed to the grouped transaction extensions:
The code is here: https://github.com/paritytech/polkadot-sdk/blob/25691b8e945a6453fa65fd5ae6eb54b6a523f280/substrate/primitives/runtime/src/traits/transaction_extension/mod.rs#L510-L533

To be more precise:
* for `(A, B), C, D)` the inherited implications of A is `ext_version, call, C_explicit, D_explicit, C_implicit, D_implicit, B_explicit, B_implicit`.
* for `(A, B, C, D)` the inherited implications of A is `ext_version, call, B_explicit, C_explicit, D_explicit, B_implicit, C_implicit, D_implicit`.
and for both the metadata is simply `A, B, C`

### Solution:

Don't use tuple, make a new struct with 200 optional generics named `TransactionExtensionPipeline`, and derive traits on it.

cc @georgepisaltu
